File: usedesk_sdk/src/main/java/ru/usedesk/sdk/data/framework/entity/request/SendFeedbackRequest.java
package ru.usedesk.sdk.data.framework.entity.request;

import ru.usedesk.sdk.domain.entity.Feedback;

import static ru.usedesk.sdk.domain.entity.Constants.VALUE_FEEDBACK_ACTION;

public class SendFeedbackRequest extends BaseRequest {

    public static final String TYPE = "@@server/chat/CALLBACK";

    private Payload payload;

    public SendFeedbackRequest(String token, Feedback feedback) {
        super(TYPE, token);
        payload = new Payload(feedback);
    }

    private class Payload {

        private String type;

        private Feedback data;

        public Payload(Feedback feedback) {
            type = VALUE_FEEDBACK_ACTION;
            this.data = feedback;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/internal/data/repository/api/_entity/request/SendFeedbackRequest.kt
package ru.usedesk.chat_sdk.internal.data.repository.api._entity.request

import ru.usedesk.chat_sdk.external.entity.UsedeskFeedback

class SendFeedbackRequest(
        token: String,
        feedback: UsedeskFeedback
) : BaseRequest(TYPE, token) {

    private val payload: Payload

    private class Payload internal constructor(feedback: UsedeskFeedback) {
        private val type: String = VALUE_FEEDBACK_ACTION

        private val data: UsedeskFeedback = feedback
    }

    companion object {
        private const val TYPE = "@@server/chat/CALLBACK"
        private const val VALUE_FEEDBACK_ACTION = "action"
    }

    init {
        payload = Payload(feedback)
    }
}